openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
+                # Not yet implemented as MGXS do not yet support this
+                pass
+
+        else:
+            if self._representation is 'isotropic':
+                shape = (self.num_orders, self._energy_groups.num_groups,
+                         self._energy_groups.num_groups)
+                max_depth = 3
+            elif self._representation is 'angle':
+                shape = (self._num_polar, self._num_azimuthal, self.num_orders,
+                         self._energy_groups.num_groups,
+                         self._energy_groups.num_groups)
+                max_depth = 5
+            # check we have a numpy list
+            check_iterable_type("scatter", scatter, expected_type=Real,
+                                max_depth=max_depth)
+            if scatter.shape == shape:
+                self._scatter = np.copy(scatter)
+            else:
+                msg = 'Shape of provided scatter "{0}" does not match shape ' \
+                      'required, "{1}"'.format(scatter.shape, shape)
+                raise ValueError(msg)
+
+    def set_multiplicity(self, multiplicity, scatter=None, **kwargs):
+        if isinstance(multiplicity, openmc.mgxs.NuScatterMatrixXS):
+            if not isinstance(scatter, openmc.mgxs.ScatterMatrixXS):
+                msg = "Argument 'scatter' must be provided."
+                raise ValueError(msg)
+            # Make sure passed MGXS objects contain correct group structure
+            if self.energy_groups != multiplicity.energy_groups:
+                msg = 'Group structure of provided NuScatterMatrixXS does not ' \
+                      'match group structure of XSdata object'
+                raise ValueError(msg)
+            if self.energy_groups != scatter.energy_groups:
+                msg = 'Group structure of provided ScatterMatrixXS does not ' \
+                      'match group structure of XSdata object'
+                raise ValueError(msg)
+            # Get openmc.mgxs.get_xs() arguments from kwargs
+            # nuclides, xs_type, and value will have sane defaults but can be
+            # overridden by kwards
+            if 'nuclides' in kwargs:
+                nuclides = kwargs['nuclides']
+            else:
+                nuclides = 'sum'
+            if 'xs_type' in kwargs:
+                xs_type = kwargs['xs_type']
+            else:
+                xs_type = 'macro'
+            if 'value' in kwargs:
+                value = kwargs['value']
+            else:
+                value = 'mean'
+            # subdomains is required from the kwargs as this is specific to
+            # this XSdata object.
+            if 'subdomains' in kwargs:
+                subdomains = kwargs['subdomains']
+            else:
+                msg = "Argument 'subdomains' is required"
+                raise ValueError(msg)
+
+            if self._representation is 'isotropic':
+                nuscatt = multiplicity.get_xs(subdomains=subdomains,
```

Comment:
use tally arithmetic to simplify:

```python
nu = multiplicity / scatter
self._multiplicity = nu.get_xs(nuclides=nuclides, xs_type=xs_type)
```

I've left out `subdomains` and `value` since I see no reason to keep them around per my earlier comments.